synapse/handlers/auth.py

```diff
@@ -1226,6 +1240,20 @@ def _complete_sso_login(
         )
         respond_with_html(request, 200, html)
 
+    async def _sso_login_callback(self, login_result: JsonDict) -> None:
+        """
+        A login callback which might add additional attributes to the login response.
+
+        Args:
+            login_result: The data to be sent to the client. Includes the user
+                ID and access token.
+        """
+        extra_attributes = self._extra_attributes.get(login_result["user_id"])
+        if extra_attributes:
+            login_result.update(
```

Comment:
TIL one can update a dict with a tuple!

Reply:
Any iterable of iterable of two elements actually. 😄